qiskit/providers/ibmq/runtime/runtime_job.py

```diff
@@ -177,12 +178,22 @@ def status(self) -> JobStatus:
         """
         if self._status not in JOB_FINAL_STATES:
             response = self._api_client.job_get(job_id=self.job_id())
+            response_job_status = response['status'].upper()
             try:
-                self._status = API_TO_JOB_STATUS[response['status'].upper()]
+                self._status = API_TO_JOB_STATUS[response_job_status]
             except KeyError:
                 raise IBMQError(f"Unknown status: {response['status']}")
+            self._set_error_message(response_job_status=response_job_status)
         return self._status
 
+    def error_message(self) -> Optional[str]:
```

Comment:
I think this method can be slightly more helpful. For example, `_error_message` is only set in `status()`, so if a user doesn't call `status()` first (e.g. they saw the status on IQX), they will only get `None. In addition, `Unable to retrieve result for job` is not super helpful. The `results` endpoint returns stderr if the job failed (see line 150), so that can be used here as well.

Reply:
Fixed!